### PR TITLE
Add vscode files

### PR DESCRIPTION
**Pull request by robredpath**
(for info, really, don't merge)

### CODE REVIEW

**Comment by robredpath** [comment]
Obviously I've used a lot of vscode scaffolding here, but all of this could probably be implemented in an IDE-agnostic way with a single Dockerfile.

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
+// README at: https://github.com/devcontainers/templates/tree/main/src/python
+{
+	"name": "Python 3",
+	// Or use a Dockerfile or Docker Compose file. More info: https://containers.dev/guide/dockerfile
+	"image": "python:3.11",
```
Comment (robredpath) [inline .devcontainer/devcontainer.json:6]: @odscjames I tried this today and it works fine for this purpose. The MS image is actually a very basic bit of stuff on top of the basic images - see https://github.com/devcontainers/images/blob/main/src/python/.devcontainer/Dockerfile

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	// Configure tool-specific properties.
+	"customizations": {
+		// Configure properties specific to VS Code.
+		"vscode": {
+			"settings": {},
+			"extensions": [
+				"ms-azuretools.vscode-docker",
+				"lextudio.restructuredtext-pack",
+				"ms-python.debugpy",
+				"streetsidesoftware.code-spell-checker"
```
Comment (robredpath) [inline .devcontainer/devcontainer.json:26]: @tillywoodfield what spell checker do you use?

Comment (robredpath) [reply .devcontainer/devcontainer.json:26]: @tillywoodfield @simon-20 I also just had a bit of a play and I can see that VSCode doesn't automatically add installed extensions to devcontainer.json so if there are extensions that we don't want everyone to have to install, we can just omit them. 

They're very easy to install inside the container via the Extensions sidebar, especially as any extensions you have installed locally are top of the list. Although they obviously don't persist between rebuilds, I wouldn't expect these containers to be getting rebuilt so often that would be problematic.

Comment (tillywoodfield) [reply .devcontainer/devcontainer.json:26]: > @tillywoodfield what spell checker do you use?

It is this one! I didn't recognise the name the other day 😄

Comment (odscjames) [reply .devcontainer/devcontainer.json:26]: English (GB) - turn on by changing "cSpell.language": "en" to "cSpell.language": "en-GB"

Is this something that should be in this config?

**Comment by robredpath** [comment]
Closing as all of the relevant work is now in [the branch where I'm actually writing docs](https://github.com/IATI/iati-validator-docs/pull/3) because, unsurprisingly, I actually needed this stuff in my working tree!

**Pull request closed without merging**